### PR TITLE
Correct booking url domain

### LINKED ISSUE #386: Show bookingUrl with correct domain to make QA happy
Clicking on "Manage my booking" button redirects to kiwi.com production page, whre you are not logged-in, which lower the user experience on staging. Let's make QA lifes better by fixing this. :heart: 

This needs to be done on GraphQL server: https://github.com/kiwicom/graphql/issues/183

### PR DESCRIPTION
**Pull request by kantord**
closes #386

<br/><br/><br/><url>LiveURL: https://smartfaq-correct-bookingURL-domain.surge.sh</url>

### CODE REVIEW

**Comment by git-toni** [comment]
Is this closing #386?

**Comment by kantord** [comment]
@git-toni yes, thanks

**Comment by git-toni** [comment]
I think it's simpler to merge this PR than to change the graphql sever as suggested by @jaroslav-kubicek. Agree Jarda?

**Comment by kantord** [comment]
oh yes, I forgot to mention, haha. I considered changing graphql as well, but it seemed much more problematic than doing this. With graphql the base domain would have to be supplied or inferred.

A mixed solution would be to add a `directAccessPath` along `directAccessURL` on GraphQL, eliminating the need to parse the URL on the frontend

**Comment by git-toni** [comment]
Agree, that was exactly what I intended to do @kantord as suggested [here](https://github.com/kiwicom/graphql/issues/183#issuecomment-396192199). And also agree that this can be a refactor later on once we really need it, so I'd go with your PR as well :slightly_smiling_face:

File: src/helpers/replaceDomain.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+// @flow
+
+import parse from 'url-parse';
+
+const replaceDomain = (URL: string, domain: string) => {
+  const parsed = parse(URL);
```
Comment (git-toni) [inline src/helpers/replaceDomain.js]: what is we `const parsed = new URL(myUrlString)`? We'd save the dependency on `url-parse`.

File: src/helpers/replaceWithCurrentDomain.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+// @flow
```
Comment (git-toni) [inline src/helpers/replaceWithCurrentDomain.js]: Can we group both helpers in one file? They seem quite related :slightly_smiling_face:

Comment (kantord) [reply src/helpers/replaceWithCurrentDomain.js]: ¡Claro! Should be cool now :wink:

**Comment by kantord** [comment]
@jaroslav-kubicek let's merge this?

File: src/helpers/replaceWithCurrentDomain.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+export const replaceDomain = (link: string, domain: string) => {
+  const parsed = new URL(link);
```
Comment (jaroslav-kubicek) [inline src/helpers/replaceWithCurrentDomain.js]: Does it work in IE? https://developer.mozilla.org/en-US/docs/Web/API/URL#Browser_compatibility

Comment (kantord) [reply src/helpers/replaceWithCurrentDomain.js]: Maybe we can polyfill it too. Or just revert to adding a dependency for this :joy: Or just using a regexp to implement a limited solution that nevertheless covers our needs.

File: src/helpers/__tests__/replaceWithCurrentDomain.test.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+import { replaceDomain } from '../replaceWithCurrentDomain';
+
+const testCases = [
+  ['https://example.com', 'https://domain.net', 'https://domain.net/'],
+  ['https://example.com', 'https://domain2.org', 'https://domain2.org/'],
+  ['https://kiwi.com/asdg/4/', 'https://kiwi.org', 'https://kiwi.org/asdg/4/'],
+];
+
+describe('replaceDomain', () => {
+  testCases.forEach(([URL, domain, result]) => {
```
Comment (jaroslav-kubicek) [inline src/helpers/__tests__/replaceWithCurrentDomain.test.js]: This test seems to be unnecessary complicated to comprehend, just passing one arg and checking the result is fine

**Comment by kantord** [comment]
> Pls also check if it’s working with older browsers, I am not sure if this is handled by babel or not

How?

File: src/helpers/replaceWithCurrentDomain.js (hunk 3 above)
Comment (kantord) [reply src/helpers/replaceWithCurrentDomain.js]: @git-toni what say you?

Comment (git-toni) [reply src/helpers/replaceWithCurrentDomain.js]: Yep only Opera Mini is missing https://caniuse.com/#search=url shouldn't be an issue, should it?

**Comment by git-toni** [comment]
Shouldn't `babel-preset-env` take care of browser compatibility?

**Comment by kantord** [comment]
Regexp implementation pushed!

**Comment by kantord** [comment]
rebased

**Comment by kantord** [comment]
Tests finally green :-)

**Pull request merged**